lab_statistics.Rmd

```diff
@@ -0,0 +1,287 @@
+range_var2 <- range(dataset$Variable_2, na.rm = TRUE)
+range_var1
+range_var2
+```
+
+For categorical variables we can count the queries.  
+```{r table-cat, accordion = T}
+table(dataset$Category_1)
+table(dataset[,c('Category_1', 'Category_2', 'Gender')])
+```
+
+
+## Interquartile. 
+Calculate the following for `Variable_1`:   
+
+- Min.  
+- Median.  
+- Max.  
+- Quantiles (q1:25% and q2:75%) by using `quantile` function.  
+- Interquartile range (q3 - q1).      
+
+**Note:** Make sure to control for missing values (`na.rm = T`)
+```{r interquartile, accordion = T}
+min_val <- min(dataset$Variable_1, na.rm = T)
+q1 <- quantile(dataset$Variable_1, 0.25, na.rm = T)
+median_val <- median(dataset$Variable_1, na.rm = T)
+q3 <- quantile(dataset$Variable_1, 0.75, na.rm = T)
+max_val <- max(dataset$Variable_1, na.rm = T)
+print('Variable_1 range:')
+c(min_val, q1, median_val, q3, max_val)
+
+# Interqurtile: 
+print('Interquartile:')
+iqr_val1 <- q3 - q1
+iqr_val1
+# or
+iqr_val1 <- IQR(dataset$Variable_1, na.rm = T)
+iqr_val1 
+```
+## Standard deviation
+- `sd` is a built-in function in R to calculate standard deviation. 
+- Calculate standard deviation of `Variable_1` for males and females separately.  
+
+**Note:** Make sure to control for missing values (`na.rm = T`)
+```{r sd, accordion = T}
+sd_var1_f <- sd(dataset_f$Variable_1, na.rm = TRUE)
+sd_var1_m <- sd(dataset_m$Variable_1, na.rm = TRUE)
+sd_var1_f
+sd_var1_m
+```
+
+## Variance
+- `var` is a built-in function in R to calculate variance.  
+- Calculate variance of `Variable_1` for males and females separately. 
+
+**Note:** Make sure to control for missing values (`na.rm = T`). 
+```{r var, accordion = T}
+var_var1_f <- var(dataset_f$Variable_1, na.rm = TRUE)
+var_var1_m <- var(dataset_m$Variable_1, na.rm = TRUE)
+var_var1_f
+var_var1_m
+```
+
+
+
+
+# Correlation  
+## Pearson's correlation  
+- Let's check the correlation of `Variable_1` and `Variable_2` by Pearson's method implemented in `cor` function. You can specify the method by `method =  'pearson'`.  
+
+**Note:** Here we handle the missing values a bit differently.  In `cor` function, you control the missing values (`NA`) by setting `use = 'complete.obs'` which means to include observations that there are values for both variables. In other words, rows with `NA` in either of the values will be excluded.   
+
+```{r pearson, accordion = T}
+pearson_correlation <- cor(dataset$Variable_1, dataset$Variable_2, use = 'complete.obs', method = "pearson")
+pearson_correlation
+```
+
+- You can also plot the values in a simple plot using `plot` function. How do you interpret the results?  
+
+```{r v1v2plot, accordion = T}
+plot(dataset$Variable_1, dataset$Variable_2, main = 'Variable_1 vs Variable_2', xlab = 'Variable_1', ylab = 'Variable_2')
+```
+
+## Spearman's correlation  
+- You can calculate Spearman's correlation using the same function (`cor`) for both numerical and categorical data.  
```

Comment:
I think the pearson correlation on categorical data that is not binary is correct if the categories could be ranked, something like "low", "medium", or "high" on dosage. But if the categories are arbitrary, I am not sure it would be a good idea to use spearman rank coef.

Double check in case I am wrong, or maybe we could specify on the exercise that the categories are ranked from A to E, or change the names of the Category_1 to make them ordinal in some way?